### PR TITLE
Add leaderboard API

### PR DESCRIPTION
**Pull request by Rishi-Mishra0704**
# Use of Leaderboard API

**_In this PR, I have applied the leaderboard API in my leaderboard project here are some features:_**

- Optimize The add and refresh button
- Add leaderboard API
- Use Async Await wherever it was in need
- Store Items in local Storage
- UI does not update unless the refresh button is clicked

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,48 +1,77 @@
...
+    const scores = JSON.parse(localStorage.getItem(gameId)) || [];
+    return { scores };
+  },
+  submitScore: async (gameId, name, score) => {
+    const scores = JSON.parse(localStorage.getItem(gameId)) || [];
+    scores.push({ name, score });
+    localStorage.setItem(gameId, JSON.stringify(scores));
+    return { scores };
+  },
+};
```
Comment (Owusu-Desmond) [inline src/index.js]: - [ ] Please from the project requirement, you should read and use the [Leaderboard API documentation](https://www.notion.so/microverse/Leaderboard-API-service-24c0c3c116974ac49488d4eb0267ade3) and make sure that you send and receive data from/to the [Leaderboard API](https://www.notion.so/microverse/Leaderboard-API-service-24c0c3c116974ac49488d4eb0267ade3) but not the `localStorage` :smile: So make sure that you use `async` and `await` with the `fetch` method to send and receive data from the Leaderboard API.

File: src/index.js [hunk 2]
```diff
@@ -1,48 +1,77 @@
 import './style.css';
+/* eslint-disable no-console */
+/* eslint-disable arrow-body-style */
```
Comment (Owusu-Desmond) [inline src/index.js]: - [ ] I dont think you should disable any of the `eslint` style guides since they enhance quality and clean code.

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @Rishi-Mishra0704 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please make sure that you remove this comment from your readme :point_down: 

  _NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Rishi-Mishra0704** [comment]
@Owusu-Desmond  Can I use the mock API that I created for createGame submitScore etc?

File: src/index.js [hunk 3]
```diff
@@ -1,48 +1,107 @@
...
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ name: gameName }),
+  });
+
+  const data = await response.json();
+
+  gameId = data.result;
+};
```
Comment (Qoosim) [inline src/index.js]: - [ ] Great job implementing the reqeust from previous reviewe.  However, there're some issues you still need to work o before your project meet all the requirements for the milestone.
-  First, when the submit button is clicked, it hits the POST API and user entries is posted. But your app behaves differently to this. Kindly look into it and fix.

### Your app behaviou
[app-behave.webm](https://user-images.githubusercontent.com/26597432/225344894-0759e0d5-154a-4f86-9d1f-489889328c88.webm)

File: src/index.js [hunk 4]
```diff
@@ -1,48 +1,107 @@
...
+  const response = await fetch(`${API_URL}/games/${gameId}/scores`);
+  const data = await response.json();
+
+  if (!data || !data.result) {
+    throw new Error('Invalid response from server');
+  }
+
+  return data.result;
+};
+
```
Comment (Qoosim) [inline src/index.js:32]: - [ ] Great job implementing the reqeust from previous reviewe.  However, there're some issues you still need to work o before your project meet all the requirements for the milestone.
-  Second, When the Refresh button is click, it should hit the GET API endpoint causing the scores list to change. But your app behaves differently to this. Kindly look into it and fix.


### Your app behaviou
[app-behave.webm](https://user-images.githubusercontent.com/26597432/225344894-0759e0d5-154a-4f86-9d1f-489889328c88.webm)

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @Rishi-Mishra0704, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 




 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (Rishi-Mishra0704) [reply src/index.js]: @Qoosim  when the refresh button is clicked , should give all the scores in the api or should I set some limit

File: src/index.js (hunk 4 above)
Comment (Rishi-Mishra0704) [reply src/index.js:32]: @Qoosim when the refresh button is clicked , should give all the scores in the api or should I set some limit

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi @Rishi-Mishra0704  🙋‍♂️, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.



## Top highlight 🎯
- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [ ] Although you have successfully made the necessary modifications, please keep in mind that clicking the Refresh button should trigger the GET API endpoint to update the scores list. It should not be done automatically. Please review and rectify this issue.


https://user-images.githubusercontent.com/105117832/225570471-b068c1dc-e3c4-4821-a7b1-32f32f395521.mp4



- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BenMukebo — APPROVED** [review]
### Hi @Rishi-Mishra0704 , 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- The submit button  hit the POST API endpoint and post user entries. ✔️ 
- The Refresh hit the GET API endpoint causing the scores list to change. ✔️  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL] As requiested by the previous reviewer, Please delete this comment  👇🏻from License section as these are instructions on how to complete the License section.

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**